docs/manual/transform_data.rst

```diff
@@ -131,6 +131,118 @@ Transformation of a column based on a condition is possible via :func:`ifelse()`
+
+   - Single column::
+
+      >>> DT[:, dt.as_type(f.integers, str)]
+         | integers
+         | str32
+      -- + --------
+       0 | 1
+       1 | 2
+       2 | 3
+       3 | 4
+      [4 rows x 1 column]
+
+- Multiple columns::
+
+      >>> DT[:, dt.as_type(f['integers', 'floats'], str)]
+         | integers  floats
+         | str32     str32
+      -- + --------  ------
+       0 | 1         10.0
+       1 | 2         11.5
+       2 | 3         12.3
+       3 | 4         -13.0
+      [4 rows x 2 columns]
+
+Values across multiple columns can be replaced with the :meth:`datatable.Frame.replace` method; this works on the entire frame, and is an in-place operation.
+
+The :meth:`datatable.Frame.replace` syntax is -> ``frame.replace(replace_what, replace_with)``, where `replace_what` values in the frame are replaced with `replaced_with`::
+
+   >>> DF = DT.copy()
+   >>> DF.replace(1, -1)
+   >>> DF
+      | dates       integers   floats  strings
+      | date32         int32  float64  str32
+   -- + ----------  --------  -------  -------
+    0 | 2000-01-05        -1     10    A
+    1 | 2010-11-23         2     11.5  B
+    2 | 2020-02-29         3     12.3  NA
+    3 | NA                 4    -13    D
+   [4 rows x 4 columns]
+
+For multiple values, a list or a dictionary can be used.
+
+If a list is used, the number of entries in the `replace_what` list must match the number of entries in the `replace_with` list, or the number of entries in `replace_with` must be exactly 1::
+
+   >>> DF.replace(['A', 10.0], ['A_pre', 30.0])
+   >>> DF
+      | dates       integers   floats  strings
+      | date32         int32  float64  str32
+   -- + ----------  --------  -------  -------
+    0 | 2000-01-05        -1     30    A_pre
+    1 | 2010-11-23         2     11.5  B
+    2 | 2020-02-29         3     12.3  NA
+    3 | NA                 4    -13    D
+   [4 rows x 4 columns]
+
+   >>> DF.replace([2, 3], 20)
+   >>> DF
+      | dates       integers   floats  strings
+      | date32         int32  float64  str32
+   -- + ----------  --------  -------  -------
+    0 | 2000-01-05        -1     30    A_pre
+    1 | 2010-11-23        20     11.5  B
+    2 | 2020-02-29        20     12.3  NA
+    3 | NA                 4    -13    D
+   [4 rows x 4 columns]
+
+If a dictionary is used, the `replace_what` values serve as the keys, while the `replace_with` values are the values in the dictionary::
+
+   >>> DF.replace({4: 24, 'B': 'BBB'})
+   >>> DF
+      | dates       integers   floats  strings
+      | date32         int32  float64  str32
+   -- + ----------  --------  -------  -------
+    0 | 2000-01-05        -1     30    A_pre
+    1 | 2010-11-23         2     11.5  BBB
+    2 | 2020-02-29         3     12.3  NA
+    3 | NA                24    -13    D
+   [4 rows x 4 columns]
+
+Note that the data type in `replace_what` and `replace_with` must match the data type of the existing column::
```

Comment:
I guess it is better to remove "the data type of the existing column" part, because the values should just be of the same type no matter which columns are present in the frame. If they both don't match the type of the column, no replacement will take place. But if they don't match each other, an error will be thrown.